libcudacxx/test/libcudacxx/std/utilities/optional/optional.object/optional.object.observe/dereference.pass.cpp

```diff
@@ -68,14 +62,43 @@ int main(int, char**)
     // operator.
     // Regardless this function should still be noexcept(false) because
     // it has a narrow contract.
+
+    optional<X&> optref;
+    unused(optref);
+    ASSERT_SAME_TYPE(decltype(*optref), X&);
+    LIBCPP_STATIC_ASSERT(noexcept(*optref), "");
+    ASSERT_NOEXCEPT(*optref);
   }
+
   {
     optional<X> opt(X{});
     assert((*opt).test() == 4);
   }
-#if !(defined(TEST_COMPILER_CUDACC_BELOW_11_3) && defined(TEST_COMPILER_CLANG))
-  static_assert(test() == 7, "");
-#endif // !(defined(TEST_COMPILER_CUDACC_BELOW_11_3) && defined(TEST_COMPILER_CLANG))
+
+  {
+    X val{};
+    optional<X&> opt(val);
+    assert((*opt).test() == 4);
```

Comment:
These and similar should verify that the address held by the optional is the same as the address it is supposed to be referring to. This applies to everything in the tests of the observers, `swap`, and probably also ctors.

Reply:
Yeah that is a good point

Reply:
I have added tests for all observers, will go through the constructors tomorrow